lib/bolt/pal/yaml_plan/evaluator.rb

```diff
@@ -174,6 +176,17 @@ def evaluate_code_blocks(scope, value)
         def evaluate(value, _scope)
           value
         end
+
+        # TODO: Remove this method once 'target' parameter is removed
+        def targets(type, step)
+          if step.key?('target')
```

Comment:
Not sure where to attach this comment, but it might be nicer to handle this warning in `evaluate_block_with_bindings` so that you only get the warning once even if multiple steps refer to `target`. We could just check `plan.steps.any? { |step| step.key?('target') }` and issue a generic warning.